frame/ranked-collective/src/lib.rs

```diff
@@ -228,16 +234,30 @@ impl Convert<Rank, Votes> for Geometric {
 }
 
 /// Trait for getting the maximum number of voters for a given rank.
-pub trait GetMaxVoters {
+pub trait MaxRankVoters {
```

Comment:
Could be a single trait, generic over some `id`, but by default for rank and class, it's the same type.
Keeping it scoped for the purpose of the pr and simple.

Reply:
I really don't see why we would want two traits which both have exactly the same interface and do exactly the same thing in this pallet since Rank === Class.

Reply:
I would just leave one impl of `MaxClassVoters` for simplicity, and use it for the impl of `VoteTally` over `Class`, not `Rank`.

Reply:
Why not just call it `GetMaxVoters` as it already is?

Reply:
@gavofyork renamed, also added some test that fails with previous version.